src/main/java/org/opensearch/neuralsearch/stats/NeuralStatsInput.java

```diff
@@ -0,0 +1,168 @@
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Holds user requester input parameters for retrieving neural stats via rest handler API.
+ * It allows filtering statistics by node IDs, event statistic types, and state statistic types.
+ */
+@Getter
+public class NeuralStatsInput implements ToXContentObject, Writeable {
+    public static final String NODE_IDS_FIELD = "node_ids";
+    public static final String EVENT_STAT_NAMES_FIELD = "event_stats";
+    public static final String STATE_STAT_NAMES_FIELD = "state_stats";
+
+    /**
+     * Collection of node IDs to filter statistics retrieval.
+     * If empty, stats from all nodes will be retrieved.
+     */
+    private Set<String> nodeIds;
+
+    /**
+     * Collection of event statistic types to filter.
+     */
+    private EnumSet<EventStatName> eventStatNames;
+
+    /**
+     * Collection of state statistic types to filter.
+     */
+    private EnumSet<StateStatName> stateStatNames;
+
+    /**
+     * Controls whether metadata should be included in the statistics response.
+     */
+    @Setter
+    private boolean includeMetadata;
+
+    /**
+     * Controls whether the response keys should be flattened.
+     */
+    @Setter
+    private boolean flatten;
+
+    /**
+     * Builder constructor for creating NeuralStatsInput with specific filtering parameters.
+     *
+     * @param nodeIds node IDs to retrieve stats from
+     * @param eventStatNames event stats to retrieve
+     * @param stateStatNames state stats to retrieve
+     * @param includeMetadata whether to include metadata
+     * @param flatten whether to flatten keys
+     */
+    @Builder
+    public NeuralStatsInput(
+        Set<String> nodeIds,
+        EnumSet<EventStatName> eventStatNames,
+        EnumSet<StateStatName> stateStatNames,
+        boolean includeMetadata,
+        boolean flatten
+    ) {
+        this.nodeIds = nodeIds;
+        this.eventStatNames = eventStatNames;
+        this.stateStatNames = stateStatNames;
+        this.includeMetadata = includeMetadata;
+        this.flatten = flatten;
+    }
+
+    /**
+     * Default constructor that initializes with empty filters and default settings.
+     * By default, metadata is excluded and keys are not flattened.
+     */
+    public NeuralStatsInput() {
+        this.nodeIds = new HashSet<>();
+        this.eventStatNames = EnumSet.noneOf(EventStatName.class);
+        this.stateStatNames = EnumSet.noneOf(StateStatName.class);
+        this.includeMetadata = false;
+        this.flatten = false;
+    }
+
+    /**
+     * Constructor for stream input
+     *
+     * @param input the StreamInput to read data from
+     * @throws IOException if there's an error reading from the stream
+     */
+    public NeuralStatsInput(StreamInput input) throws IOException {
+        nodeIds = input.readBoolean() ? new HashSet<>(input.readStringList()) : new HashSet<>();
```

Comment:
`readOptionalStringList();` might be better in terms of readability.
Wonder if we can make nodeIds as List so that we can avoid this list to set conversion.